deeppavlov/models/go_bot/network.py

```diff
@@ -499,12 +499,15 @@ def reset(self):
             log.debug("Bot reset.")
 
     def destroy(self):
-        if callable(getattr(self.slot_filler, 'destroy', None)):
-            self.slot_filler.destroy()
-        if callable(getattr(self.embedder, 'destroy', None)):
-            self.embedder.destroy()
-        if callable(getattr(self.intent_classifier, 'destroy', None)):
-            self.intent_classifier.destroy()
+        if hasattr(self, 'slot_filler'):
+            if callable(getattr(self.slot_filler, 'destroy', None)):
+                self.slot_filler.destroy()
+        if hasattr(self, 'embedder'):
+            if callable(getattr(self.embedder, 'destroy', None)):
+                self.embedder.destroy()
+        if hasattr(self, 'intent_classifier'):
+            if callable(getattr(self.intent_classifier, 'destroy', None)):
+                self.intent_classifier.destroy()
```

Comment:
Isn't this implementation redundant? Destroy should be called for every attribute in `Component.destroy()` anyway